pkg/splunk/util/secrets.go

```diff
@@ -495,14 +495,13 @@ func ApplyNamespaceScopedSecretObject(ctx context.Context, client splcommon.Cont
 }
 
 // GetSecretByName retrieves namespace scoped secret object for a given name
-func GetSecretByName(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, name string) (*corev1.Secret, error) {
+func GetSecretByName(ctx context.Context, c splcommon.ControllerClient, namespace string, logHandle string, name string) (*corev1.Secret, error) {
```

Comment:
Why did we change the arguments? Earlier approach of passing the cr & extracting the Name & NS in one place within GetSecretByName seemed a cleaner approach (rather than in all callers)

Reply:
See [here](https://github.com/splunk/splunk-operator/pull/730#discussion_r848975369). The API prototype was changed as the purpose of the API is to retrieve a secret by its name, namespace(has nothing to do with **splcommon.MetaObject**). This helps in making the API being accessible to areas of code where there are no objects of type splcommon.MetaObject involved.